mllib/src/test/scala/org/apache/spark/ml/tuning/CrossValidatorSuite.scala

```diff
@@ -40,10 +41,22 @@ class CrossValidatorSuite
   import testImplicits._
 
   @transient var dataset: Dataset[_] = _
+  @transient var datasetWithFold: Dataset[_] = _
 
   override def beforeAll(): Unit = {
     super.beforeAll()
     dataset = sc.parallelize(generateLogisticInput(1.0, 1.0, 100, 42), 2).toDF()
+    val foldCol = udf { () =>
+      val r = Math.random()
```

Comment:
Not sure if it matters but do you want to create this list ahead of time from a seeded RNG, and parallelize it, to ensure it doesn't vary?

Reply:
Good idea.